src/EventGridExtension/EventHubCapturePublisher.cs

```diff
@@ -75,43 +90,31 @@ public List<IDisposable> Recycles
                 else
                 {
                     // convert from stream 
-                    HttpWebRequest myHttpWebRequest = (HttpWebRequest)WebRequest.Create(data.FileUrl);
+                    var blobStream = await blob.OpenReadAsync();
                     if (t == typeof(Stream))
                     {
                         _recycles = new List<IDisposable>();
-                        // SHUN TODO async
-                        HttpWebResponse myHttpWebResponse = (HttpWebResponse)myHttpWebRequest.GetResponse();
-                        Stream responseStream = myHttpWebResponse.GetResponseStream();
-                        _recycles.Add(responseStream);
-                        _recycles.Add(myHttpWebResponse);
-
-                        bindingData.Add("EventGridTrigger", responseStream);
+                        _recycles.Add(blobStream); // close after function call
+                        bindingData.Add("EventGridTrigger", blobStream);
                     }
                     // copy to memory => use case javascript
                     else if (t == typeof(Byte[]))
                     {
-                        using (HttpWebResponse myHttpWebResponse = (HttpWebResponse)myHttpWebRequest.GetResponse())
+                        using (MemoryStream ms = new MemoryStream())
```

Comment:
Its not urgent, but you should chat with @MikeStall about if there is a way you can get this logic to reuse our standard blob binding logic. Right now every time we add a new type that you can bind to a blob, you would have to update this code too.